src/instructlab/model/evaluate.py

```diff
@@ -33,889 +32,768 @@ class Benchmark(str, enum.Enum):
-                    model_name,
-                    judge_model_name,
-                    taxonomy_path,
-                    branch,
-                    output_dir,
-                    merge_system_user_message=merge_system_user_message,
-                ),
-                MTBenchBranchEvaluator(
-                    base_model_name,
-                    judge_model_name,
-                    taxonomy_path,
-                    base_branch,
-                    output_dir,
-                    merge_system_user_message=merge_system_user_message,
-                ),
-            ]
-            branches = [branch, base_branch]
-            m_paths = [model, base_model]
-            m_names = [model_name, base_model_name]
-            qa_pairs_and_errors = []
-            server = None
+        validate_model(model, allow_gguf=False)
+        if benchmark == Benchmark.MMLU_BRANCH:
+            validate_model(base_model, "--base-model", allow_gguf=False)
 
-            for i, evaluator in enumerate(evaluators):
-                branch = branches[i]
-                m_path = m_paths[i]
-                m_name = m_names[i]
 
-                print(
-                    f"Generating questions and reference answers from qna files for branch {branch}..."
+def validate_model(model: str, model_arg: str = "--model", allow_gguf: bool = True):
+    if os.path.exists(model):
+        model_path = pathlib.Path(model)
+        valid_model = False
+        if model_path.is_dir():
+            valid_model = is_model_safetensors(model_path)
+        elif model_path.is_file():
+            if allow_gguf:
+                valid_model = is_model_gguf(model_path)
+            else:
+                click.secho(
+                    "MMLU and MMLUBranch can currently only be used with a safetensors directory",
+                    fg="red",
                 )
-                try:
-                    server, api_base, effective_gpus = launch_server(
-                        eval_serve=ctx.obj.config.serve,
-                        tls_client_cert=ctx.params["tls_client_cert"],
-                        tls_client_key=ctx.params["tls_client_key"],
-                        tls_client_passwd=ctx.params["tls_client_passwd"],
-                        tls_insecure=ctx.params["tls_insecure"],
-                        model=m_path,
-                        model_name=m_name,
-                        max_workers=max_workers,
-                        gpus=gpus,
-                        backend=backend,
-                        enable_serving_output=enable_serving_output,
-                    )
-                    evaluator.gen_answers(
-                        api_base, max_workers=max_workers, serving_gpus=effective_gpus
-                    )
-                finally:
-                    if server is not None:
-                        server.shutdown()
+                raise click.exceptions.Exit(1)
+        if not valid_model:
+            click.secho(
```

Comment:
please replace all instances of `click.secho` with `logger.info` or general exception raises